p3387.tex

```diff
@@ -239,7 +245,8 @@ \section{Design goals}
 
 \subsection{Allow \tcode{pre} and \tcode{post} on coroutines}
 
-The simplest and most practical design goal of this proposal is to allow applying \tcode{pre} and \tcode{post} to coroutines in some form. This allows us to specify contract assertions on the parameters passed into the coroutine when it is called, assertions on the object that this function call returns to the caller at the point of first suspension, as well as assertions on other aspects of the state of the program at those points.
+The simplest and most practical design goal of this proposal is to allow applying \tcode{pre} and \tcode{post} to coroutines in some form. This allows us to specify contract assertions on the parameters passed into the coroutine when it is called, assertions on the object that this function call returns to the caller, ???as well as assertions on other aspects of the state of the program at those points.
+%IDS is this meant to refer to use of contract_assert (if so, then maybe move it to a new place, since it's already a done-deal and need not impact on design decisions here).
```

Comment:
no, it's about pre/post that refer to global state independent of the parametesr and return value.

Reply:
OK - perhaps a small amendment to clarify this would help.

Reply:
Maybe like this?
"This allows us to specify precondition assertions on the parameters passed into the coroutine, postcondition assertions on the object that this function call returns to the caller, as well as preconditions and postconditions on other reachable program state at the point when the coroutine is called and when it returns."
It's hard to formulate this better, but if you have an idea, please let me know!

Reply:
that seems OK - I now understand the intent.